File: Tracko-Flutter-UI/lib/models/account.dart
import 'dart:async';

import 'package:tracko/models/transaction.dart';
import 'package:tracko/models/user.dart';

class Account {
  int? id;
  String name = '';
  int? userId;
  String currency = 'INR';
  List<Transaction> transactions = [];

  Account();

  Account.make(this.id, this.name, this.userId, {this.currency = 'INR'});

  @override
  String toString() {
    return 'Account{id: $id, name: $name, userId: $userId, currency: $currency, transactions: $transactions}';
  }

  static int defaultAccountId() {
    return 1;
  }
}

File: backend/src/main/java/com/trako/entities/Account.java
package com.trako.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
@Entity
@Table(name = "accounts")
@Getter
@Setter
public class Account {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @NotNull
    @Column(name = "name", length = 250)
    private String name;

    @NotNull
    @Column(name = "user_id", length = 36)
    private String userId;

    @Column(name = "currency", length = 3)
    private String currency = "INR";

    @Transient
    private Double balance;

    @JsonIgnore
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", insertable = false, updatable = false)
    private User user;
}
